src/lib/security/selinux_config.rb

```diff
@@ -0,0 +1,263 @@
+      end
+
+      self.mode
+    end
+
+    # Returns the configured SELinux mode according to params in kernel command line
+    #
+    # @see #mode_from_kernel_params
+    #
+    # @return [Symbol] the mode identifier
+    def configured_mode
+      Mode.match(mode_from_kernel_params)
+    end
+
+    # Returns the SELinux configuration based on options set in the kernel command line
+    #
+    # @return [Symbol] the mode identifier
+    def mode_from_kernel_params
+      params = Mode.keys.flat_map do |key|
+        value = Array(read_param(key)).last
+        next if value == :missing
+
+        [key, value]
+      end
+
+      Hash[*params.compact]
+    end
+
+    # Convenience method to read a value from kernel parameters
+    #
+    # @see Yast::Bootloader#kernel_param
+    def read_param(key)
+      Yast::Bootloader.kernel_param(:common, key.to_s)
+    end
+
+    # Model that represents a SELinux mode
+    class Mode
+      extend Yast::I18n
+      include Yast::I18n
+
+      # @return [Symbol] the id representing the mode
+      attr_reader :id
+      alias_method :to_sym, :id
+
+      # @return [String] the human readable name to represent the mode
+      attr_reader :name
+      alias_method :to_human_string, :name
+
+      # @return [Hash{String=><String, :missing>}] options for setting the mode via kernel params
+      attr_reader :options
+
+      # Returns all known SELinux modes
+      #
+      # @return [Array<Mode>]
+      def self.all
+        ALL.dup
+      end
+
+      # Returns all known keys for setting a SELinux mode via kernel command line
+      #
+      # @return [Array<String>]
+      def self.keys
+        OPTIONS_KEYS
+      end
+
+      # Finds a SELinux mode by its id
+      #
+      # @return [Mode, nil] found mode or nil if given mode id does not exists
+      def self.find(id)
+        ALL.find { |mode| mode.id == id.to_sym }
+      end
+
+      # Finds the SELinux mode which fits better to given options
+      #
+      # @return [Mode] proper mode according to values of given options
+      def self.match(options)
+        return find(:disabled) if options.empty?
+
+        security_module = options["security"]
+        module_disabled = options["selinux"].to_i <= 0
+        enforcing_mode  = options["enforcing"].to_i > 0
+
+        return find(:disabled) if security_module != "selinux" || module_disabled
+        return find(:permissive) unless enforcing_mode
+
+        find(:enforcing)
+      end
+
+      # Constructor
+      #
+      # Intended to be used internally by the class
+      #
+      # @param id [String, Symbol] if of the mode
+      # @param name [String] the mode name, a string marked for translation
+      # @param enable [Boolean] wether the mode will enable SELinux
+      # @param enforcing [Boolean] if SELinux should be run enforcing or not
+      def initialize(id, name, enable, enforcing)
+        textdomain "security"
+
+        @id = id.to_sym
+        @name = _(name)
```

Comment:
I worry this cannot be done. As during proposal user can change language then we need also strings retranslated. So instead of hardcoded translation I would store here marked name and have method
```
def to_human_string
  _(@name)
end
```

This way it always return string respecting *current* language. Common pitfall in installation when language can change easily.

Reply:
Good to know. I didn't realize that it is _kind of_ hard-coded translation. Thanks

Reply:
well, it works that `_()` return localized string and you remember it. So if later proposal change language it is still memorized.